### DISCUSSION TITLE
LFE channel in RTMP is missing

### DISCUSSION THREAD

**Discussion by rsarkisov82** [ant-media/Ant-Media-Server#6343] [Q&A]
Hi, 

We have another issue I need your help with
I can't get audio in 4th channel neither if I monitor the source in ANTmedia dashboard not over AMS players.
The source is RTMP with 6 or 8ch audio.
I may assume that for some reason the 4ch is being recognized as an LFE channel and LFE is on.
Anyway, we have no issues with the same input on other receivers.

- Roman

**Comment by Mohit-3196** [comment]
Hi @rsarkisov82,
How are you?

Can you please check the audio mapping with ffmpeg and share the result?
`ffmpeg -i rtmp://your_stream_url -map 0:a -c copy -f null -
`

Also, if you can share your Ant Media Server log file with us along with the streamId and the timeframe, I can check the logs to see if there are any warnings or errors about audio processing with Ant Media Server.

As a temporary workaround, if possible, reconfigure the input stream to exclude the problematic channel or remap it using FFmpeg before sending it to the Ant Media Server.
`ffmpeg -i rtmp://your_stream_url -map_channel 0.1.3 -c:a copy -f flv rtmp://antmedia_server_url
` 
I hope it helps.

--
Mohit

**Comment by rsarkisov82** [comment]
Hi Mohit,

I've run ffmpeg as a server in listening mode and this is the result:
[rtmp @ 000001deba4bdbc0] App field don't match up: WebRTCAppEE/ <-> WebRTCAppEE
[aac @ 000001debad4e7c0] This stream seems to incorrectly report its last channel as SCE[1], mapping to LFE[0]
Input #0, flv, from 'rtmp://100.103.214.209:1935/WebRTCAppEE/MGWL':
  Metadata:
    copyright       : default
    fileSize        : 0
    audiochannels   : 6
  Duration: 00:00:00.00, start: 0.818000, bitrate: N/A
  Stream #0:0: Video: h264 (High), yuv420p(tv, progressive), 1920x1080, 8388 kb/s, 60 fps, 59.94 tbr, 1k tbn
  Stream #0:1: Audio: aac (LC), 48000 Hz, 5.1, fltp, 131 kb/s
Output #0, null, to 'pipe:':
  Metadata:
    copyright       : default
    fileSize        : 0
    audiochannels   : 6
    encoder         : Lavf60.11.100
  Stream #0:0: Audio: aac (LC), 48000 Hz, 5.1, fltp, 131 kb/s
Stream mapping:
  Stream #0:1 -> #0:0 (copy)
Press [q] to stop, [?] for help
size=N/A time=00:00:04.83 bitrate=N/A speed=1.18x    s speed=N/A

**Comment by Mohit-3196** [comment]
Hi @rsarkisov82,
Thank you.

Did you try the workaround solution as well?

Thank you.

**Comment by rsarkisov82** [comment]
Hi Mohit,
no, I didn't. even it it works, it's not an option for us as it would significantly increased delay. 
We replaced the encoders for now, and waiting if there is anything you can do with it.
- Roman

**Reply by Mohit-3196** [reply]
Hi @rsarkisov82,
Thank you.
Can you please share a recording from your encoder output?
You can do that by using this ffmpeg command and pushing a stream

```
rtmp://localhost:1935/live.
ffmpeg -listen 1 -i rtmp://localhost:1935/live -c copy output.mp4
```

Thank you.

**Comment by rsarkisov82** [comment]
ffmpeg -f flv -listen 1 -i rtmp://100.103.214.209:1935/live/app -c copy c:/output.mp4   gives me this 

[rtmp @ 000001ba4278da40] App field don't match up: WebRTCAppEE/ <-> live
[rtmp @ 000001ba4278da40] Unexpected stream MGWL, expecting app
    Last message repeated 1 times
[aac @ 000001ba4303dac0] This stream seems to incorrectly report its last channel as SCE[1], mapping to LFE[0]
Input #0, flv, from 'rtmp://100.103.214.209:1935/live/app':
  Metadata:
    copyright       : default
    fileSize        : 0
    audiochannels   : 6
  Duration: 00:00:00.00, start: 0.919000, bitrate: N/A
  Stream #0:0: Video: h264 (High), yuv420p(tv, progressive), 1920x1080, 8388 kb/s, 60 fps, 59.94 tbr, 1k tbn
  Stream #0:1: Audio: aac (LC), 48000 Hz, 5.1, fltp, 131 kb/s
File 'c:/output.mp4' already exists. Overwrite? [y/N] y
Output #0, mp4, to 'c:/output.mp4':
  Metadata:
    copyright       : default
    fileSize        : 0
    audiochannels   : 6
    encoder         : Lavf60.11.100
  Stream #0:0: Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, progressive), 1920x1080, q=2-31, 8388 kb/s, 60 fps, 59.94 tbr, 16k tbn
  Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, 5.1, fltp, 131 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
[out#0/mp4 @ 000001ba4364eec0] video:9229kB audio:282kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.277266%
frame= 1076 fps= 84 q=-1.0 Lsize=    9537kB time=00:00:17.96 bitrate=4350.0kbits/s speed=1.41x
Exiting normally, received signal 2.
